darwin/DarwinProcess.c

```diff
@@ -368,31 +383,61 @@ void DarwinProcess_scanThreads(DarwinProcess* dp) {
    integer_t run_state = 999;
    for (unsigned int i = 0; i < thread_count; i++) {
       thread_info_data_t thinfo;
-      mach_msg_type_number_t thread_info_count = THREAD_BASIC_INFO_COUNT;
-      ret = thread_info(thread_list[i], THREAD_BASIC_INFO, (thread_info_t)thinfo, &thread_info_count);
+      mach_msg_type_number_t thread_info_count = THREAD_EXTENDED_INFO_COUNT;
+      ret = thread_info(thread_list[i], THREAD_EXTENDED_INFO_COUNT, (thread_info_t)thinfo, &thread_info_count);
```

Comment:
are you sure that the `thinfo` struct is passed by copy and not by pointer-to? I've seen a couple of SO posts passing it in via `(thread_info_t) &thinfo`: https://stackoverflow.com/a/33414855/4583130

the same actually applies to the similar logic above for obtaining the `TASK_BASIC_INFO`.